### DISCUSSION TITLE
Duvida Sobre javascript Objetos e funções em determinado projeto

### DISCUSSION THREAD

**Discussion by evertonProgramadorCriativo** [frontendbr/forum#1624] [Perguntas e Respostas]
link --->https://www.w3schools.com/w3css/w3css_slideshow.asp

JavaScript explicado
Primeiro, defina o slideIndex como 1. (Primeira imagem)

Em seguida, chame showDivs () para exibir a primeira imagem.

Quando o usuário clica em um dos botões, chama plusDivs () .

A função plusDivs () subtrai um ou  adiciona um ao slideIndex.

A função showDiv () oculta ( display = "none" ) todos os elementos com o nome da classe "mySlides" e exibe ( display = "block" ) o elemento com o slideIndex fornecido.

Se o slideIndex for maior que o número de elementos (x.length), o slideIndex será definido como zero.

Se o slideIndex for menor que 1, ele será definido para o número de elementos (comprimento x).

/--------------------------------------------------------/

/---------------------Pergunta--01----------------/

/--------------------------------------------------------/

function showDivs(n) {...} ela esta escutando a function plusDivs(n) {...} e

var slideIndex = 1;  Resumo a function showDivs(n) {...} ela foi passada como um objeto ou como uma  classList ?

/--------------------------------------------------------/

/---------------------Pergunta--02----------------/

/--------------------------------------------------------/

A plusDivs(n) {...}  por padrao ela valer (n) que no caso e 0 , quando o botao for clicado ela (ADICIONAR+1 ou SUBTRAI -1) uma posicao da foto(mySlides) RESUMO DA PERGUNTA , AS DUAS FUNCAO ESTAO COM O MESMO NOME NO PARAMETRO , ESSE PARAMETRO ESTA INTERLIGADO AS DUAS OU CADA FUNCAO  TEM O SEU PARAMETRO INDIVIDUAL ?

function showDivs(n) {...} 

function plusDivs(n) {...}

/--------------------------------------------------------/

/---------------------Pergunta--03----------------/

/--------------------------------------------------------/

O for esta passando display = "none"; para todas as class foto(mySlides)  como ele consegue associar a mudancao de evento onclick  ? Por que assim que ele for clicado o index da imagem fica display em bloco 

for (i = 0; i < x.length; i++) {

    x[i].style.display = "none";  

  }

/--------------------------------------Referencia da pergunta ---------------/

<!DOCTYPE html>

<html>

<title>W3.CSS</title>

<meta name="viewport" content="width=device-width, initial-scale=1">

<link rel="stylesheet" href="https://www.w3schools.com/w3css/4/w3.css">

<style>

.mySlides {display:none;}

</style>

<body>

<h2 class="w3-center">Manual Slideshow</h2>

<div class="w3-content w3-display-container">

  <img class="mySlides" src="img_snowtops.jpg" style="width:100%">

  <img class="mySlides" src="img_lights.jpg" style="width:100%">

  <img class="mySlides" src="img_mountains.jpg" style="width:100%">

  <img class="mySlides" src="img_forest.jpg" style="width:100%">

  <button class="w3-button w3-black w3-display-left" onclick="plusDivs(-1)">&#10094;</button>

  <button class="w3-button w3-black w3-display-right" onclick="plusDivs(1)">&#10095;</button>

</div>

<script>

var slideIndex = 1;

showDivs(slideIndex);

function plusDivs(n) {

  showDivs(slideIndex += n);

}

function showDivs(n) {

  var i;

  var x = document.getElementsByClassName("mySlides");

  if (n > x.length) {slideIndex = 1}

  if (n < 1) {slideIndex = x.length}

  for (i = 0; i < x.length; i++) {

    x[i].style.display = "none";  

  }

  x[slideIndex-1].style.display = "block";  

}

</script>

</body>

**Comment by developerfred** [comment]
Fala @evertonProgramadorCriativo Tudo bem? a diagramação da sua pergunta esta bem confusa. te aconselho a da uma olhada neste artigo do @fdaciuk  [Aprenda markdown](https://blog.da2k.com.br/2015/02/08/aprenda-markdown/)

**Comment by eliseumds** [comment]
Coloca isso aí num CodeSandBox da vida, ou CodePen. E o título tá mto chamativo, os mods não vão curtir.
